### DISCUSSION TITLE
Saving model weights before nan

### DISCUSSION THREAD

**Discussion by marcofrancis** [lululxvi/deepxde#1013] [Q&A] [answered]
Hi,
I'm training my model (pytorch backend) usong adam for 10K steps and then BFGS.
BFGS stops because it hits some nans. How can I recover the model state before it hit the nans?
Is there a way to save the last state before nans?

**Comment by lululxvi** [comment, marked as answer]
Use `dde.callbacks.ModelCheckpoint` https://deepxde.readthedocs.io/en/latest/modules/deepxde.html?highlight=modelcheck#deepxde.callbacks.ModelCheckpoint
